core/specfem/receivers/dim2/receiver.cpp

```
@@ -17,3 +17,12 @@ specfem::receivers::receiver<specfem::dimension::type::dim2>::print() const {
 
   return message.str();
 }
+
+bool specfem::receivers::receiver<specfem::dimension::type::dim2>::operator==(
+    const receiver &other) const {
+  return (this->network_name == other.network_name) &&
+         (this->station_name == other.station_name) &&
+         specfem::utilities::is_close(this->x, other.x) &&
+         specfem::utilities::is_close(this->z, other.z) &&
+         specfem::utilities::is_close(this->angle, other.angle);
+}
```

Can 2 receivers have same station and network name but have different locations/angle? Should this be a assert?

---

I think the components of a receiver (angle) can definitely be different but the names should be the same as the station.

This is really only used for checking the read-in stations in the unit tests, I'm not sure whether there is a different use case. If you just want to check whether receivers share the same station name or just the same location, the approach should be slightly different maybe implementing a new a method like `same_code(other)` or `same_location(other)`

We could implement a function that weeds out potential duplicates, or throws an error if a vector of receivers contains two receivers of the same angle and location.

---

No I am wondering we should throw a runtime error if stations have same name but different location/angle. It's more to do with symantics of operator==. For example, what would be the expected behavior when someone uses the function outside of test.

---

Oh, understood! Yes i agree.